open_optical_gating/cli/cli.py

```diff
@@ -95,26 +114,33 @@ def initialise_internal_parameters(self):
 
         # Sets ouput mode for normal trigger or special Glasgow mode
         if self.settings["trigger_mode"] == "5V_BNC_Only":
-            self.trigger_mode = 1
+            self.moveStagesAfterTrigger = False
         else:
-            self.trigger_mode = 0
-
-        # Initialises reference frames if not specified
-        # state has several modes:
-        #   0 - run prospective gating mode (phase locked triggering)
-        #   1 - re-initialise (clears for mode 2)
-        #   2 - get period mode (requires user input)
-        #   3 - adaptive mode (update period but maintain phase lock)
+            self.moveStagesAfterTrigger = True
+
+        # TODO: JT writes: this seems as good a place as any to flag the fact that I don't think barrier frames are being implemented properly.
+        # There is a call to determine_barrier_frames, but I don't think the *value* for the barrier frame parameter is ever computed, is it?
+        # It certainly isn't when using existing reference frames. This seems like an important missing bit of code.
+        # I think it just defaults to 0 when the settings are initialised, and stays that way.

```

Comment:
So... well spotted... I remember discovering this myself a few months back... but forgetting about it.

Reply:
Opened issue; resolving this conversation